kernel/sched.c

```diff
@@ -1237,14 +1237,18 @@ static inline void z_vrfy_k_yield(void)
 #include <syscalls/k_yield_mrsh.c>
 #endif
 
-static int32_t z_tick_sleep(int32_t ticks)
+static int32_t z_tick_sleep(k_ticks_t ticks)
 {
 #ifdef CONFIG_MULTITHREADING
 	uint32_t expected_wakeup_time;
 
 	__ASSERT(!arch_is_in_isr(), "");
 
-	LOG_DBG("thread %p for %d ticks", _current, ticks);
+#ifdef CONFIG_TIMEOUT_64BIT
+	LOG_DBG("thread %p for %lld ticks", _current, ticks);
```

Comment:
This will select the correct format specifier independent of the system.
```suggestion
	LOG_DBG("thread %p for " PRId64 " ticks", _current, ticks);
```